src/core/QueryBuilderMacros.scala

```diff
@@ -110,72 +96,101 @@ class QueryBuilderMacros(val c: blackbox.Context) {
   }
 
   def orderByImpl[T: c.WeakTypeTag](pred: c.Tree*): c.universe.Tree = {
-    val conditions = pred.collect {
-      case q"(..$_) => $_.asc((..$_) => $select)" => true -> disassembleSelectTree(select)
-      case q"(..$_) => $_.desc((..$_) => $select)" =>
-        false -> disassembleSelectTree(select)
-    }
-
-    val sortBy = conditions.map {
-      case (isAscending, dt) =>
-        if (isAscending) {
-          q"_root_.com.google.cloud.datastore.StructuredQuery.OrderBy.asc(${dt.pathLiteral})"
-        } else {
-          q"_root_.com.google.cloud.datastore.StructuredQuery.OrderBy.desc(${dt.pathLiteral})"
-        }
-    }
+    val sortBy: Seq[c.Tree] = pred
+      .collect {
+        case q"(..$_) => $_.asc((..$_) => $select)"  => true -> CallTree(select)
+        case q"(..$_) => $_.desc((..$_) => $select)" => false -> CallTree(select)
+      }
+      .flatMap {
+        case (isAscending, ct) =>
+          ct.resolveCriteria.map {
+            case AppliedCriteria(path, _, _) =>
+              if (isAscending) {
+                q"_root_.com.google.cloud.datastore.StructuredQuery.OrderBy.asc($path)"
+              } else {
+                q"_root_.com.google.cloud.datastore.StructuredQuery.OrderBy.desc($path)"
+              }
+          }
+      }
     q"$self.withSortCriteria(..$sortBy)"
   }
 
-  type OpArgs = (c.TermName, List[c.Tree])
-  private case class DisassembledTree(path: String, operations: List[OpArgs]) {
-    val pathLiteral: c.universe.Literal = Literal(Constant(path))
-  }
+  case class CallTree(tree: mutatus.utils.BinaryTree[c.Tree]) {
+    def resolveCriteria: List[AppliedCriteria] = {
+      def buildPath(tree: c.Tree, prefix: String = ""): String = {
+        val path = tree.toString().indexOf(".") match {
```

Comment:
You don't need to change this, but `tree.toString.split("\.")` might be more natural.

Reply:
However, we maybe shouldn't do the `toString`. Technically, it's an unlikely scenario, but what if there's a value such as,
```
val `this` = ...
val `super` = ...
```

As soon as we convert to `String`s we lose some useful information. I don't think that should stop us merging, but can you include a `FIXME` in the code, and maybe open an issue mentioning this?